src/fixup.js

```diff
@@ -141,6 +141,29 @@
     }
   }
 
+  /* fragment identifiers backward compatibility mapping */
+  if (location.hash &&
+    !document.getElementById(location.hash.substr(1)) && // not broken link
```

Comment:
nit: if we use `!document.querySelector(location.hash)` we don't need to get the substring.